pulsar/consumer_partition.go

```diff
@@ -417,6 +417,93 @@ func (pc *partitionConsumer) Unsubscribe() error {
 	return req.err
 }
 
+func (pc *partitionConsumer) AckIDWithTxn(msgID MessageID, txn Transaction) error {
```

Comment:
Could you reduce the duplicated code with `ackID`? I see much code is just copied from `ackID`.